rasa/nlu/emulators/luis.py

```diff
@@ -1,50 +1,89 @@
-from typing import Any, Dict, Text
+from typing import Any, Dict, List, Text
 
 from rasa.nlu.emulators.emulator import Emulator
-from typing import List, Optional
+from rasa.shared.nlu.constants import (
+    ENTITIES,
+    ENTITY_ATTRIBUTE_TYPE,
+    ENTITY_ATTRIBUTE_ROLE,
+    ENTITY_ATTRIBUTE_VALUE,
+    ENTITY_ATTRIBUTE_START,
+    ENTITY_ATTRIBUTE_END,
+    EXTRACTOR,
+    INTENT_RANKING_KEY,
+    TEXT,
+    INTENT,
+    INTENT_NAME_KEY,
+    PREDICTED_CONFIDENCE_KEY,
+)
 
 
 class LUISEmulator(Emulator):
-    """Emulates Luis responses."""
+    """Emulates the response format of the LUIS Endpoint API v3.0 /predict endpoint.
 
-    def _top_intent(self, data) -> Optional[Dict[Text, Any]]:
-        if data.get("intent"):
+    https://westcentralus.dev.cognitive.microsoft.com/docs/services/luis-endpoint-api-v3-0/
+    https://docs.microsoft.com/en-us/azure/cognitive-services/LUIS/luis-concept-data-extraction?tabs=V3
+    """
+
+    def _intents(self, data: Dict[Text, Any]) -> Dict[Text, Any]:
+        if data.get(INTENT_RANKING_KEY):
             return {
-                "intent": data["intent"]["name"],
-                "score": data["intent"]["confidence"],
+                intent[INTENT_NAME_KEY]: {"score": intent[PREDICTED_CONFIDENCE_KEY]}
+                for intent in data[INTENT_RANKING_KEY]
             }
-        else:
-            return None
-
-    def _ranking(self, data) -> List[Dict[Text, Any]]:
-        if data.get("intent_ranking"):
-            return [
-                {"intent": el["name"], "score": el["confidence"]}
-                for el in data["intent_ranking"]
-            ]
-        else:
-            top = self._top_intent(data)
-            return [top] if top else []
 
-    def normalise_response_json(self, data: Dict[Text, Any]) -> Dict[Text, Any]:
-        """Transform data to luis.ai format."""
+        top = data.get(INTENT)
+        if not top:
+            return {}
 
-        top_intent = self._top_intent(data)
-        ranking = self._ranking(data)
-        return {
-            "query": data["text"],
-            "topScoringIntent": top_intent,
-            "intents": ranking,
-            "entities": [
+        return {top[INTENT_NAME_KEY]: {"score": top[PREDICTED_CONFIDENCE_KEY]}}
+
+    def _entities(
+        self, data: Dict[Text, Any]
+    ) -> Dict[Text, Dict[Text, List[Dict[Text, Any]]]]:
+        if ENTITIES not in data:
+            return {}
+
+        entities = {"$instance": {}}
+        for e in data[ENTITIES]:
+            # LUIS API v3 uses entity roles instead of entity names
+            # (it's possible because its roles are unique):
+            # https://docs.microsoft.com/en-us/azure/cognitive-services/LUIS/luis-migration-api-v3#entity-role-name-instead-of-entity-name
+            key = e.get(ENTITY_ATTRIBUTE_ROLE, e[ENTITY_ATTRIBUTE_TYPE])
+
+            entities.update({key: [e[ENTITY_ATTRIBUTE_VALUE]]})
```

Comment:
optional: seems a bit easier too read for me
```suggestion
            entities[key] = [e[ENTITY_ATTRIBUTE_VALUE]]
```